page-component-builder/src/components/ComponentPalette/index.js

```diff
@@ -92,20 +92,35 @@ const ComponentPalette = props => {
             </div>
           </div>
           <div className="component-palette__collection">
-            {props.schemas
-              .filter(schema => schema.meta.editor_role === c.PATTERN_ROLE)
-              .map((schema, index) => {
-                return (
-                  <Card
-                    key={`palette-schema--${index}`}
-                    testId="palette-schema"
-                    className="component-palette__schema"
-                    selected={schema.meta.id === props.componentId}
-                    onClick={() => handleSchemaClick(schema)}>
-                    {schema.meta.id || 'Schema'}
-                  </Card>
-                );
-              })}
+            {selectedTags.length
+              ? props.schemas
+                  .filter(schema => schema.meta.tags.some(tag => selectedTags.includes(tag)))
+                  .map((schema, index) => {
+                    return (
+                      <Card
+                        key={`palette-schema--${index}`}
+                        testId="palette-schema"
+                        className="component-palette__schema"
+                        selected={schema.meta.id === props.componentId}
+                        onClick={() => handleSchemaClick(schema)}>
+                        {schema.meta.id || 'Schema'}
+                      </Card>
+                    );
+                  })
+              : props.schemas
+                  .filter(schema => schema.meta.editor_role === c.PATTERN_ROLE)
+                  .map((schema, index) => {
+                    return (
+                      <Card
+                        key={`palette-schema--${index}`}
+                        testId="palette-schema"
+                        className="component-palette__schema"
+                        selected={schema.meta.id === props.componentId}
+                        onClick={() => handleSchemaClick(schema)}>
+                        {schema.meta.id || 'Schema'}
+                      </Card>
+                    );
+                  })}
```

Comment:
try to `DRY` up ("Don't Repeat Yourself") the code so you're only calling `.map` a single time (since it's the exact same `.map` call for each. 

Maybe if the logic is getting a bit heavy (especially since you're adding the text filter soon) it'll be good to refactor part of the logic into its own function.

i'd recommend maybe refactoring everything before the `.map` into a function which returns exactly what's being filtered and then calling `.map` on it. 

for example:

```
filteredSchemes(...args) {
  <logic>
  return schemas
}

render() {
  {
    filteredSchemas().map((schema, index) => {
      return (
                      <Card
                        key={`palette-schema--${index}`}
                        testId="palette-schema"
                        className="component-palette__schema"
                        selected={schema.meta.id === props.componentId}
                        onClick={() => handleSchemaClick(schema)}>
                        {schema.meta.id || 'Schema'}
                      </Card>
                    );
    })
  }
}
```